packages/cli/src/commands/rollback/index.ts

```diff
@@ -74,48 +73,49 @@ export default async (client: Client): Promise<number> => {
     return 2;
   }
 
-  // ensure the current directory is good
-  const cwd = argv['--cwd'] || process.cwd();
-  const pathValidation = await validatePaths(client, [cwd]);
-  if (!pathValidation.valid) {
-    return pathValidation.exitCode;
-  }
-
-  // ensure the current directory is a linked project
-  const linkedProject = await ensureLink(
-    'rollback',
-    client,
-    pathValidation.path,
-    {
-      autoConfirm: Boolean(argv['--yes']),
-    }
-  );
-  if (typeof linkedProject === 'number') {
-    return linkedProject;
-  }
-
   // validate the timeout
   let timeout = argv['--timeout'];
   if (timeout && ms(timeout) === undefined) {
     client.output.error(`Invalid timeout "${timeout}"`);
     return 1;
   }
 
-  const { project } = linkedProject;
   const actionOrDeployId = argv._[1] || 'status';
 
-  if (actionOrDeployId === 'status') {
-    return await rollbackStatus({
+  try {
+    if (actionOrDeployId === 'status') {
+      const project = await getProjectByCwdOrLink({
+        autoConfirm: Boolean(argv['--yes']),
+        client,
+        commandName: 'promote',
+        cwd: argv['--cwd'] || process.cwd(),
+        projectNameOrId: argv._[2],
+      });
+
+      return await rollbackStatus({
+        client,
+        project,
+        timeout,
+      });
+    }
+
+    return await requestRollback({
       client,
-      project,
+      deployId: actionOrDeployId,
       timeout,
     });
-  }
+  } catch (err) {
+    if (isErrnoException(err)) {
+      if (err.code === 'ERR_CANCELED') {
+        return 0;
+      }
+      if (err.code === 'ERR_INVALID_CWD' || err.code === 'ERR_LINK_PROJECT') {
+        // do not show the message
```

Comment:
No need, `validatePaths()` already prints an error: https://github.com/vercel/vercel/blob/main/packages/cli/src/util/validate-paths.ts#L72. These error codes are in theory temporary until we refactor the code to throw instead of returning the exit code from functions.